[#31] 팔로우 / 언팔로우 기능 구현

cyj199637 opened the pull request:
- 팔로우 기능 구현
- 언팔로우 기능 구현
- 팔로우 시 알림 보내기 기능 구현

cyj199637 committed:
[#31] 팔로우 / 언팔로우 기능 구현

- 팔로우 기능 구현
- 언팔로우 기능 구현

cyj199637 committed:
Merge branch 'feature/29' into feature/31

cyj199637 committed:
[#31] 알림 추가 기능 구현

- 팔로잉된 사용자에게 알림이 가도록 FollowController의 followUser()에 알림 추가 기능을 추가

cyj199637 committed:
[#31] Follow 테이블 컬럼 변경된 이름 반영

f-lab-dev commented on src/main/java/me/liiot/snsserver/controller/FollowController.java:
```diff
@@ -0,0 +1,43 @@
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/follow")
```
리소스를 나타낼때 이 부분은 복수형이여야합니다~ 이 부분은 문서를 조금 더 읽어보시면 좋을 것 같습니다

f-lab-dev commented on src/main/java/me/liiot/snsserver/controller/FollowController.java:
```diff
@@ -0,0 +1,43 @@
+                                           @CurrentUser User currentUser) {
+
+        followService.addFollowList(currentUser.getUserId(), followUserId);
+        alarmService.addAlarm(currentUser.getUserId(), followUserId, AlarmType.FOLLOWING);
```
**팔로우를 하면 알림을 보낸다**는 service layer의 책임일까요? presentation layer의 책임일까요?

cyj199637 replied:
팔로우를 하면 알림을 보낸다는 것은 일종의 비즈니스 로직이기 때문에 Service Layer에서 처리하는게 맞다고 생각합니다. 수정하겠습니다!

f-lab-dev commented on src/main/java/me/liiot/snsserver/service/AlarmServiceImpl.java:
```diff
@@ -0,0 +1,22 @@
+    private final AlarmMapper alarmMapper;
+
+    @Override
+    public void addAlarm(String userId, String targetId, AlarmType type) {
```
이 이슈의 제목은 팔로우 / 언팔로우인데 알람도 같이 구현하셨네요~ push 알림쪽은 파이어베이스쪽 한번 봐보시면 좋을 것 같아요~

f-lab-dev commented on src/main/java/me/liiot/snsserver/service/FollowService.java:
```diff
@@ -0,0 +1,8 @@
+package me.liiot.snsserver.service;
+
+public interface FollowService {
```
이 인터페이스는 어떤 쪽으로의 확장을 염두에 두신건가요? 이런 확장 시나리오는 주석으로 남겨두시면 좋을 것 같습니다

cyj199637 replied:
처음에는 나중에 추가되는 기능이나 비즈니스 로직에 따라 Follow에 관한 기능들의 로직이 변경될 것 같아
인터페이스를 두었습니다. 하지만 다시 고민해보니 구현체를 따로 둬야 할 만큼의 큰 변화가 예상되지 않아서
일단 FollowService 인터페이스는 제거했습니다.

cyj199637 committed:
[#31] 피드백 반영

- FollowController URL 수정
- FollowService에서 알림 기능을 처리하도록 수정
- FollowService 인터페이스 제거

cyj199637 committed:
[#37] Firebase를 활용한 푸시 서비스 (#38)

* [#37] Firebase를 활용한 푸시 서비스

- Firebase를 활용하여 푸시 서비스 구현
- 푸시 메시지 전송을 비동기로 처리

* [#37] Firebase 푸시 서비스 수정

- 각 클래스마다 사용한 어노테이션에 대한 주석 추가
- AccessToken을 Redis에 저장하여 관리하도록 로직 변경

* [#37] Firebase 푸시 서비스 수정

- 로그인 / 로그아웃에 AccessToken 등록 / 삭제 로직 추가

* [#37] Firebase 푸시 서비스 빠트린 주석 추가 및 테스트 코드 수정

* [#37] 피드백 반영

- MessageSource와 LocaleResolver를 이용하여 사용자의 위치에 따라 푸시 메시지의 언어가 다르게 나오도록 수정
- Redis에 저장되는 토큰 만료시간 설정
- 팩토리 메소드를 사용하여 스레드풀 생성

* [#37] 피드백 반영

- @Async 대신 sendAsync() 메소드로 비동기 처리

cyj199637 committed:
Merge branch 'feature/29' into feature/31
